crypto/ml_dsa/ml_dsa_encoders.c

```diff
@@ -768,7 +766,13 @@ int ossl_ml_dsa_sk_decode(ML_DSA_KEY *key, const uint8_t *in, size_t in_len)
     size_t i, k = params->k, l = params->l;
     PACKET pkt;
 
-    if (key->priv_encoding != NULL || key->pub_encoding != NULL)
+    /* When loading from an explicit key, drop the seed. */
+    OPENSSL_free(key->seed);
```

Comment:
Why cant there be both?

Reply:
raised in the issue - so closing for now..

Reply:
Because if we did not load the seed, we have no idea that it corresponds to the key we did use.